### PR TITLE
Miguel user permissions

### PR DESCRIPTION
**Pull request by migfernandes01**
New user permissions system. Centralized way to give user roles permissions. 2 new user roles (Owner and Mentor)

Document that goes deeper into the new user permission system:
https://docs.google.com/document/d/1rzCvlu5FzIXoUpWIM07Z5g80UgqN2DikT2Csz1FdtV8/edit

### CODE REVIEW

File: src/utilities/permissions.js [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    'getInvIdInfo',
+    'putInvById',
+    'getInvTypeById',
+    'putInvType',
+    'getAllInvType',
+    'postInvType',
+    'getWeeklySummaries',
+    'getTimeZoneAPIKey',
+    'checkLeadTeamOfXplus',
+  ],
```
Comment (EvianTan) [inline src/utilities/permissions.js:50]: Based on the user class document, `Core Team` seem is missing here.
image

**Comment by EvianTan** [comment]
@migfernandes01 You really did terrific work! Other than the only concern above, everything looks good to me!

**Comment by migfernandes01** [comment]
@EvianTan Thanks Yiyun, about the issues described:
- See all users in the leaderboard: it's not a permission on the code itself. There is a pipeline in MongoDB that assigns all users in the system to Admins, Owners and Core Team members, that's why there's no permission for that on the code.
- See weekly summary reports: that's a permission on the routing on react that for each directory accepts an array of allowed roles, that's why that permission is not in the permissions file.
Let me know if you understand what I said, I can also send you some screenshots if you need!

**Comment by EvianTan** [comment]
> @EvianTan Thanks Yiyun, about the issues described:
> 
> * See all users in the leaderboard: it's not a permission on the code itself. There is a pipeline in MongoDB that assigns all users in the system to Admins, Owners and Core Team members, that's why there's no permission for that on the code.
> * See weekly summary reports: that's a permission on the routing on react that for each directory accepts an array of allowed roles, that's why that permission is not in the permissions file.
>   Let me know if you understand what I said, I can also send you some screenshots if you need!

Ohhhh I understand that! I ignored it's the DB side role, thank you for the explanation!

**Review by EvianTan — APPROVED** [review]

**Pull request merged**